### PR TITLE
feat(init): support installing headers on generic linux distributions

### PR DESCRIPTION
**Pull request by dalehamel**
@leodido @fntlnz please take a look

This should enhance the `--fetch-headers` code path to work for any linux distribution, not just container OS. 

I've also optimized a bit, by deleting abundant non-header files that aren't needed, which speeds up the inter-device move pretty substantially. I also decided it would be prudent to keep track of the time spent in the major operations in the script.

I've tested the generic path by commenting out the container OS one, but please try it for yourself to be sure it's all working as expected.

### CODE REVIEW

File: init/fetch-linux-headers.sh [hunk 1]
```diff
@@ -14,6 +14,9 @@ generate_headers()
   zcat /proc/config.gz > .config
   make ARCH=x86 oldconfig > /dev/null
   make ARCH=x86 prepare > /dev/null
+
+  # Clean up abundant non-header files to speed-up copying
+  find ${BUILD_DIR} -regex '.*\.c\|.*\.txt\|.*Makefile\|.*Build\|.*Kconfig' -type f -delete
```
Comment (leodido) [inline init/fetch-linux-headers.sh]: Probably I'd prefer something like `find -name "..." -type f ...`

Comment (dalehamel) [reply init/fetch-linux-headers.sh]: Unfortunately alpine uses busybox find so I was limited to using the regex flag

Comment (leodido) [reply init/fetch-linux-headers.sh]: Ah right! Nevermind.

Comment (leodido) [reply init/fetch-linux-headers.sh]: BTW, just thinking around at 4AM ... Tomorrow I'll try with something like
`ls | grep -E "..." | xargs rm`

Comment (dalehamel) [reply init/fetch-linux-headers.sh]: > 4AM

Get some sleep ;)

**Pull request merged**